// botbuilder-ai/lib/luisV2-models/luisParameters.js
"use strict";
/**
 * @module botbuilder-ai
 */
/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.verbose = exports.timezoneOffset = exports.staging = exports.spellCheck = exports.log = exports.endpoint = exports.bingSpellCheckSubscriptionKey = exports.appId = void 0;
exports.appId = {
    parameterPath: 'appId',
    mapper: {
        required: true,
        serializedName: 'appId',
        type: {
            name: 'String',
        },
    },
};
exports.bingSpellCheckSubscriptionKey = {
    parameterPath: ['options', 'bingSpellCheckSubscriptionKey'],
    mapper: {
        serializedName: 'bing-spell-check-subscription-key',
        type: {
            name: 'String',
        },
    },
};
exports.endpoint = {
    parameterPath: 'endpoint',
    mapper: {
        required: true,
        serializedName: 'Endpoint',
        defaultValue: '',
        type: {
            name: 'String',
        },
    },
    skipEncoding: true,
};
exports.log = {
    parameterPath: ['options', 'log'],
    mapper: {
        serializedName: 'log',
        type: {
            name: 'Boolean',
        },
    },
};
exports.spellCheck = {
    parameterPath: ['options', 'spellCheck'],
    mapper: {
        serializedName: 'spellCheck',
        type: {
            name: 'Boolean',
        },
    },
};
exports.staging = {
    parameterPath: ['options', 'staging'],
    mapper: {
        serializedName: 'staging',
        type: {
            name: 'Boolean',
        },
    },
};
exports.timezoneOffset = {
    parameterPath: ['options', 'timezoneOffset'],
    mapper: {
        serializedName: 'timezoneOffset',
        type: {
            name: 'Number',
        },
    },
};
exports.verbose = {
    parameterPath: ['options', 'verbose'],
    mapper: {
        serializedName: 'verbose',
        type: {
            name: 'Boolean',
        },
    },
};
